src/sklearn_evaluation/plot/precision_recall.py

```diff
@@ -4,148 +4,332 @@
 from sklearn.preprocessing import label_binarize
 from ..telemetry import SKLearnEvaluationLogger
 from ..util import is_column_vector, is_row_vector
+from sklearn_evaluation.plot.plot import AbstractComposedPlot, AbstractPlot
+from sklearn_evaluation import __version__
 
+from warnings import warn  # noqa
 
-@SKLearnEvaluationLogger.log(feature="plot")
-def precision_recall(y_true, y_score, ax=None):
-    """
-    Plot precision-recall curve.
 
-    Parameters
-    ----------
-    y_true : array-like, shape = [n_samples]
-        Correct target values (ground truth).
-    y_score : array-like, shape = [n_samples] or [n_samples, 2] for binary
-              classification or [n_samples, n_classes] for multiclass
+def _set_ax_settings(ax, name):
+    ax.set_title(name)
+    ax.set_xlim([0.0, 1.0])
+    ax.set_ylim([0.0, 1.05])
+    ax.set_xlabel("Recall")
+    ax.set_ylabel("Precision")
+    ax.legend(loc="best")
 
-        Target scores (estimator predictions).
-    ax : matplotlib Axes
-        Axes object to draw the plot onto, otherwise uses current Axes
 
-    Notes
-    -----
-    It is assumed that the y_score parameter columns are in order. For example,
-    if ``y_true = [2, 2, 1, 0, 0, 1, 2]``, then the first column in y_score
-    must contain the scores for class 0, second column for class 1 and so on.
+def _validate_raw_input(y_true, y_score):
+    if any((val is None for val in (y_true, y_score))):
+        raise ValueError("y_true and y_score are needed to plot Precision-Recall")
 
 
-    Returns
-    -------
-    ax: matplotlib Axes
-        Axes containing the plot
+def _validate_metrics_input(precision, recall, average_precision):
+    if any((val is None for val in (precision, recall, average_precision))):
+        raise ValueError(
+            "precision, recall, and average_precision are needed to plot Precision-Recall"
+        )
+
+    # validate input metrics of same length
+    if len(precision) != len(recall):
+        raise ValueError("precision and recall arrays should be of same length")
```

Comment:
On the errors it will be useful for the users to understand what went wrong (true for other errors as well), so probably, print the length of each, `precision:x, recall: y`